src/pages/index.js

```diff
@@ -0,0 +1,318 @@
+  previewModal: document.querySelector("#preview-modal"),
+  previewImage: document.querySelector(".modal__image"),
+  previewCaption: document.querySelector(".modal__caption"),
+  avatarModal: document.querySelector("#avatar-modal"),
+
+  avatarPic: document.querySelector("#avatar-pic"),
+};
+
+// ------------------- API INSTANCE -------------------
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d19feaee-a024-4f2d-b0a9-cad223e76830",
+    "Content-Type": "application/json",
+  },
+});
+
+// ------------------- FETCH & RENDER DATA -------------------
+api
+  .getAppInfo()
+  .then(([userInfo, cards]) => {
+    console.log("Cards fetched:", cards);
+    selectors.profileName.textContent = userInfo.name;
+    selectors.profileDescription.textContent = userInfo.about;
+    selectors.avatarPic.src = userInfo.avatar;
+    cards.forEach((card) => renderCard(card, "append"));
+  })
+  .catch(console.error);
+
+// ------------------- FORM HANDLERS -------------------
+const resetFormAndButton = (form) => {
+  form.reset();
+  const inputList = Array.from(form.querySelectorAll(settings.inputSelector));
+  const submitButton = form.querySelector(settings.submitButtonSelector);
+  toggleButtonState(inputList, submitButton, settings);
+  resetValidation(form, settings);
+};
+
+const handleAvatarFormSubmit = (evt) => {
+  evt.preventDefault();
+  const avatarInput = document.querySelector("#profile-avatar-input").value;
+  const saveButton = avatarForm.querySelector(".modal__submit-btn");
+  saveButton.textContent = "Saving...";
+
+  api
+    .updateUserAvatar({ avatar: avatarInput })
+    .then((updatedUserInfo) => {
+      selectors.avatarPic.src = updatedUserInfo.avatar;
+      closeModal(selectors.avatarModal);
+      resetFormAndButton(avatarForm);
+    })
+    .catch((err) => console.error("Error updating avatar:", err))
+    .finally(() => {
+      saveButton.textContent = "Save";
+      saveButton.disabled = true; // Disable the button until input is valid again
+    });
+};
+
+const handleProfileFormSubmit = (evt) => {
+  evt.preventDefault();
+  const saveButton = document.querySelector(".modal__submit-btn");
+  saveButton.textContent = "Saving...";
+
+  api
+    .editUserInfo({
+      name: selectors.nameInput.value,
+      about: selectors.jobInput.value,
+    })
+    .then((updatedUserInfo) => {
+      selectors.profileName.textContent = updatedUserInfo.name;
+      selectors.profileDescription.textContent = updatedUserInfo.about;
+      closeModal(selectors.editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      saveButton.textContent = "Save";
+    });
+};
+
+const handleCardFormSubmit = (evt) => {
+  evt.preventDefault();
+  const cardName = selectors.cardForm.querySelector(
+    "#add-card-name-input"
+  ).value;
+  const cardLink = selectors.cardForm.querySelector(
+    "#add-card-link-input"
```

Comment:
* It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine